src/Http/Controllers/ImageEmbeddingController.php

```diff
@@ -38,21 +39,30 @@ public function store(Request $request, $id)
 
         $disk = Storage::disk(config('magic_sam.embedding_storage_disk'));
         $filename = "{$image->id}.npy";
-        $url = null;
+        $embedding = null;
         if ($disk->exists($filename)) {
-            if ($disk->providesTemporaryUrls()) {
-                $url = $disk->temporaryUrl($filename, now()->addHour());
-            } else {
-                $url = $disk->url($filename);
-            }
+            $path = $disk->path($filename);
+                $embedding = base64_encode(File::get($path));
         } else {
-            Queue::connection(config('magic_sam.request_connection'))
+            $job_count = config('magic_sam.job_count_cache_key');
+            if (!Cache::has($job_count)) {
+                Cache::add($job_count, 1);
+            }
+
+            if (Cache::get($job_count) > config('magic_sam.queue_threshold')) {
+                Queue::connection(config('magic_sam.request_connection'))
                 ->pushOn(
                     config('magic_sam.request_queue'),
-                    new GenerateEmbedding($image, $request->user())
-                );
+                    new GenerateEmbedding($image, $request->user()));
+                   $embedding = null;
+            } else {
+                $job = new GenerateEmbedding($image, $request->user(), False);
+                $job->handle();
+                $path = $disk->path($filename);
+                $embedding = base64_encode(File::get($path));
+            }
         }
 
-        return ['url' => $url];
+        return response()->json(['embedding' => $embedding]);
```

Comment:
Have you tried this?
```suggestion
        return response()->download($path);
```
The `response.body` should be a [Blob](https://developer.mozilla.org/en-US/docs/Web/API/Blob) then.

Reply:
When I try to use `response.body.arrayBuffer()` or `response.arrayBuffer()`, I get the following error:  `TypeError: res.(body).arrayBuffer is not a function.` I need the arrayBuffer so that npyjs can parse the embedding.

Reply:
It should work like this:
```js
EmbeddingApi.save({id: 1}, {}, {responseType: 'blob'})
   .then(r => r.body.arrayBuffer())
   .then(a => npy.parse(a));
```

Reply:
When I try:

 ```js
            ImageEmbeddingApi.save({ id: this.image.id }, {}, {responseType: 'blob'})
                .then((res) => res.body.arrayBuffer())
                .then((res) => 
                    this.handleSamEmbeddingRequestSuccess(res),
                    this.handleSamEmbeddingRequestFailure
                )
                .catch(handleErrorResponse);
```

I get this error: 
`Error in callback for watcher "isMagicSamming": "Expected up to 2 arguments [params, body], got 3 arguments"`.

Reply:
I works when you change the http method from save to get, but there is still an issue with the incorrect arrayBuffer.

Reply:
So there is still the same issue with the incorrect byte? Then we leave this issue for later...